test/patterns/eureka/eureka.chpl

```diff
@@ -0,0 +1,93 @@
+// This is a pattern for doing parallel eureka, basically:
+//   coforall across per-locale array chunks {
+//     coforall across per-CPU array chunks {
+//       search the appropriate chunk; broadcast 'Eureka!' if we find it
+//     }
+//   }
+use BlockDist, Time;
+
+// Print timing info?
+config const printTiming = false;
+
+// This is the problem size.
+config const n = 1_000_000;
+
+// This is the array of values to search in.
+const valsD: domain(1) dmapped Block(boundingBox={0..#n}) = {0..#n};
+var vals: [valsD] int;
+[i in valsD] { vals[i] = i; }
+
+// This is the value to search for.
+config const findVal = n - 1;
+
+// If we do find findVal, this is where we'll record its index.
+const foundD: domain(1) dmapped Block(boundingBox={0..#numLocales})
+              = {0..#numLocales};
+var found: [foundD] atomic int;
+[f in found] f.write(-1); // alas, can't (yet) write this as an initial value
+
+const startTime = getCurrentTime();
```

Comment:
I have a pretty strong preference for using `Timer` instead of `getCurrentTime()` -- https://github.com/chapel-lang/chapel/pull/12440#discussion_r261815638